R/deploy-site.R

```diff
@@ -0,0 +1,91 @@
+#' Deploy a pkgdown site on Travis-CI to Github Pages
+#'
+#' `deploy_site_github()` sets up your SSH keys for deployment, builds the
+#' site with [build_site()], commits the site to the `gh-pages` branch and then pushes
+#' the results back to GitHub. `deploy_site_github()` is meant only to be used
+#' by the CI system on Travis, it should not be called locally. See 'Setup' for
+#' details on setting up your repository to use this.
+#'
+#' @section Setup:
+#' Add the following to your `.travis.yml` file.
+#'
+#' ```
+#' before_deploy: Rscript -e 'install.packages("pkgdown")'
+#' deploy:
+#'   provider: script
+#'   script: Rscript -e 'pkgdown::deploy_site_github()'
+#' ```
+#'
+#' Then you will need to setup your deployment keys. The easiest way is to call
+#' `travis::use_travis_deploy()`. This will generate and push the necessary
```

Comment:
This functionality will be moved to _usethis_, https://github.com/r-lib/usethis/pull/411.